dev/partkey.md

```diff
@@ -0,0 +1,174 @@
+---
+numbersections: true
+title: "Algorand Key Specification"
+date: \today
+abstract: >
+    This document specifies the list of keys and their capabilities in Algorand.
+---
+
+# Algorand Key Specification
+
+## Overview
+
+An algorand node interacts with three types of cryptographic keys:
+
+ - _root keys_, a key pair (public and private) used to control the access to a particular account.
+   These key pairs are also known as spending keys.
+
+ - _voting keys_, a set of keys used for authentication, i.e. identify an 
+    account. Algorand uses a hierarchical (two\-level) signature scheme that 
+    ensures [forward security](https://en.wikipedia.org/wiki/Forward_secrecy), 
+    which will be detailed in next section. 
+
+ - _VRF selection keys_, keys used for proving membership of selection. 
+
+An agreement vote message is valid only when it contains a proper VRF proof and
+is signed with the correct voting key.
+
+## Root Keys
+
+Root keys are used to identify ownership of an account. An algorand node only interacts with
+the public key of a root key. The public key of a root key is also used as the account
+address. Root keys are used to sign transaction messages as well as delegating the
+voting authentication using _voting keys_, unless that specific account was rekeyed. A rekeyed
+account would use the rekeyed key in lieu of the root key.
+
+## Voting Keys 
+
+\newcommand \KeyDilution {\mathrm{KeyDilution}}
+\newcommand \Batch {\mathrm{Batch}}
+\newcommand \Offset {\mathrm{Offset}}
+
+### Algorand's Two\-level Ephemeral Signature Scheme for Authentication
+
+An ephemeral subkey is a key pair that produces one\-time signature 
+for messages. It must be deleted after use to ensure forward security.
+Algorand's ephemeral subkeys 
+uses [Ed25519 public\-key signature system](https://ed25519.cr.yp.to/).
+
+Algorand uses a two\-level ephemeral signature scheme.
+Instead of signing voting messages directly, Algorand accounts use their
+_voting keys_ to sign an intermediate ephemeral sub-key. 
+This intermediate ephemeral sub-key signs a batch of leaf-level ephemeral 
+sub-keys. Hence, each intermediate ephemeral sub-key is associated with a
+batch number ($\Batch$), and each leaf ephemeral sub-key is associate with a
+batch number (of its parent key) and an offset ($\Offset$, denotes its offset
+within a batch). A voting message is signed hierarchically: 
+the root key $\rightarrow$ batch sub\-key $\rightarrow$ leaf sub-key
```

Comment:
"root key" has a different meaning here from above: key at the root of the tree vs spending key.
And later it looks like "master key" is used with this meaning.